packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3758,6 +3758,7 @@ export class TransactionController extends BaseController<
       {
         transactionId,
         note: 'TransactionController#updateSimulationData - Update simulation data',
+        skipHistory: this.isHistoryDisabled,
```

Comment:
Rather than duplicating this check here, could we just move the check inside `#updateTransactionInternal` and have it default to use `isHistoryDisabled` if `skipHistory` is `undefined` since it's optional?

This would mean we could centralise the knowledge in one place, and also remove the `skipHistory` property inside `updateTransaction`?

Reply:
That make sense Matthew, updated